File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        i, j = 0, 0
        m, n = len(word1), len(word2)

        flag = True
        ans = ""
        while i < m and j < n:
            if flag:
                ans += word1[i]
                i += 1
            else:
                ans += word2[j]
                j += 1
            flag = not flag

        if i != m:
            ans += word1[i:]
        elif j != n:
            ans += word2[j:]
        
        return ans

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();
        int i=0, j=0;
        string ans = "";
        while(i<n && j<m) {
            ans.push_back(word1[i]);
            ans.push_back(word2[j]);
            i++;
            j++;
        }
        if(j!=0){
            while(j<m) {
                ans.push_back(word2[j]);
                j++;
            }
        }
        if(i!=0){
            while(i<n){
                ans.push_back(word1[i]);
                i++;
            }
        }
        return ans;
    }
};